src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientFactory.cs

```diff
@@ -2,21 +2,32 @@
 // Licensed under the MIT license.
 //
 using Azure.Core;
-using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal class ConfigurationClientFactory : IConfigurationClientFactory
```

Comment:
This is not a `ConfigurationClientFactory` anymore. It's a `FailOverSupportedConfigurationClientFactory`. You created `ConfigurationClient` instances in `FailOverSupportedConfigurationClient` inline without calling the Factory class.

I am wondering how much value to keep this Factory class. At least I don't see the need for the `IConfigurationClientFactory` interface.

Reply:
If as per [this comment](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/292#discussion_r839141438), we decide to use a simple `ConfigurationClient` then I see a value of having this interface and the factory class as well.

Reply:
It's still the `FailOverSupportedConfigurationClient`. Just internally, it will short circuit when there is nothing to fail over. The `FailOverSupportedConfigurationClient` uses `ConfigurationClient` internally either way.

Reply:
I was thinking we have another type `SimpleConfigurationClient` that extends `IConfigurationClient` and we return that type when the user uses `connection_string`.

Reply:
I got rid of the `IConfigurationClientFactory`.